src/Tribe/Linked_Posts.php

```diff
@@ -902,49 +906,110 @@ public function handle_submission_by_post_type( $event_id, $linked_post_type, $s
 			return;
 		}
 
+		$post_ids_to_link = $this->maybe_get_new_order_from_blocks( $event_id, $linked_post_type, $post_ids_to_link );
+		$this->maybe_reorder_linked_posts_ids( $event_id, $linked_post_type, $post_ids_to_link, $prior_linked_posts );
+	}
+
+	/**
+	 * Re-orders linked posts if the order has changed.
+	 *
+	 * @since TBD
+	 *
+	 * @param int    $event_id Event ID.
+	 * @param string $linked_post_type The post type of the linked post.
+	 * @param array  $new_order The new order of the linked posts.
+	 * @param array  $old_order The old order of the linked posts.
+	 *
+	 * @return bool
+	 */
+	public function maybe_reorder_linked_posts_ids( int $event_id, string $linked_post_type, array $new_order = [], array $old_order = [] ): bool {
```

Comment:
If you are reviewing, be sure to hide whitespace. This method is essentially the same bit of logic that was up above, but I moved it to its own method so that it could be reused. I purposefully did not refactor it to minimize changes. It isn't horrendous, luckily.